client/app/src/main/java/com/example/criengine/MyBooksActivity.java

```diff
@@ -64,6 +52,26 @@ protected void onCreate(Bundle savedInstanceState) {
         bookNameTextView = findViewById(R.id.bookListView);
         bookNameTextView.setAdapter(myBooksListAdapter);
 
+        // TODO: please pass this activity a profile object somehow, so we dont need to do this nightmare double call
```

Comment:
We would need to pass the profile object from the login screen but if there is another activity down the road that would need a profile object, are we expecting to be passing down this one object the entire way?

I feel like the double call is the better alternative.

Reply:
Yeah unfortunately I agree, at least for now.  Passing a profile through each activity wouldn't work at the moment.  However, I can look into it if I have some free time.

Reply:
I know it might not work elegantly but it might be the case that we should store profile data locally. There might be some firebase magic I can do but if not, storing the profile would cut down load time by half on this one and on requested books. Even if it's just a global. Ill keep the todo and we can look at it later